src/app/frontend/common/autorefresh/autorefresh_directive.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {AutoRefreshController} from './autorefresh_controller';
+import {DEFAULT_DELAY} from './autorefresh_controller';
+/**
+ * Provides upload file mechanism and updates the model with file parameter for the selected file.
+ * @param {!./poll_service.PollService} kdPoll
+ * @return {!angular.Directive}
+ * @ngInject
+ */
+export default function autoRefreshDirective(kdPoll) {
+  return {
+    scope: {},
+    bindToController: {
+      'source': '=',
+      'target': '=',
+      'type': '@?',
+      'delay': '@?',
+    },
+    controller: AutoRefreshController,
+    controllerAs: 'ctrl',
+    templateUrl: '',
+    link: function(scope, element, attrs) {
+      /**
+       * @type {./poll_service.PollService}
+       * @private
+       */
+      let kdPoll_ = kdPoll;
+
+      // Cancel all polls in case of state change to avoid accumulated polls.
+      /** @type {function()} */
+      let stateChangeEvent = scope.$root.$on('$stateChangeStart', () => {
+        kdPoll_.stopAll();
+      });
+      scope.$on('$destroy', stateChangeEvent);
```

Comment:
I can't catch $mdMenuOpen event in scope. I need to remove it on scope.$destroy to avoid triggering in unregistered pages as well. 
For $stateChangeStart I will fix it as sope.$on and $destroy event won't be needed for it because it is removed with the scope already. 

Any other suggestion for mdMenuOpen ?